src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -84,6 +88,44 @@ public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] Function
             }
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
+            if (function != null && function.Metadata.FunctionContainsInputBinding(WarmUpConstants.WarmupTrigger))
+            {
+                try
+                {
+                    var hostStatus = new HostStatus
+                    {
+                        State = scriptHostManager.State.ToString(),
+                        Version = ScriptHost.Version,
+                        VersionDetails = Utility.GetInformationalVersion(typeof(ScriptHost)),
+                        Id = await hostIdProvider.GetHostIdAsync(CancellationToken.None)
+                    };
+
+                    ParameterDescriptor inputParameter = function.Parameters.First(p => p.IsTrigger);
+                    Dictionary<string, object> arguments = new Dictionary<string, object>()
+                    {
+                        { inputParameter.Name, JsonConvert.SerializeObject(hostStatus) }
```

Comment:
Sending data to functions as `string` as that seemed simplest.

Lmk if there's concerns around it. In the extension, I will add a converter for string to JObject just in case if users want that.